File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(vector<int> temp:commands) {
        
        vector<int> ret;
        for(int i=temp[0]-1;i<=temp[1]-1;i++) {
            ret.push_back(array[i]);
        }
        sort(ret.begin(),ret.end());
        answer.push_back(ret[temp[2]-1]);
    }
    
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] array, int[][] commands) {
        List<Integer> ret = new ArrayList<>();
        
        for(int i=0;i<commands.length;i++) {
            List<Integer> temp = new ArrayList<>();
            for(int j=commands[i][0]-1;j<commands[i][1];j++) {
                temp.add(array[j]);
            }
            Collections.sort(temp);
            ret.add(temp.get(commands[i][2]-1));
        }
        return ret;
    }
}